File: finanze/application/ports/virtual_import_registry.py
import abc
from typing import Optional
from uuid import UUID

from domain.entity import Feature
from domain.virtual_data import VirtualDataImport, VirtualDataSource


class VirtualImportRegistry(metaclass=abc.ABCMeta):
    @abc.abstractmethod
    async def insert(self, entries: list[VirtualDataImport]):
        raise NotImplementedError

    @abc.abstractmethod
    async def get_last_import_records(
        self, source: Optional[VirtualDataSource] = None
    ) -> list[VirtualDataImport]:
        raise NotImplementedError

    @abc.abstractmethod
    async def delete_by_import_and_feature(self, import_id: UUID, feature: Feature):
        raise NotImplementedError

    @abc.abstractmethod
    async def delete_by_import_feature_and_entity(
        self, import_id: UUID, feature: Feature, entity_id: UUID
    ):
        raise NotImplementedError

File: mobile/src/application/ports/core/virtualImportRegistry.ts
import { Feature, VirtualDataImport, VirtualDataSource } from "@/domain"

export interface VirtualImportRegistry {
  insert(entries: VirtualDataImport[]): Promise<void>
  getLastImportRecords(
    source?: VirtualDataSource | null,
  ): Promise<VirtualDataImport[]>
  deleteByImportAndFeature(importId: string, feature: Feature): Promise<void>
  deleteByImportFeatureAndEntity(
    importId: string,
    feature: Feature,
    entityId: string,
  ): Promise<void>
}
